Helidon maven archetypes fail with maven-archetype-plugin version 3.3.0

## Environment Details
* Helidon Version: 2.x, 3.x, 4.x
* Helidon SE and Helidon MP

----------

## Problem Description

Running a legacy Helidon maven archetype to generate a project fails.

```
mvn -U archetype:generate -DinteractiveMode=false \
    -DarchetypeGroupId=io.helidon.archetypes \
    -DarchetypeArtifactId=helidon-quickstart-mp \
    -DarchetypeVersion=4.1.1 \
    -DgroupId=io.helidon.examples \
    -DartifactId=helidon-quickstart-mp \
    -Dpackage=io.helidon.examples.quickstart.mp

. . .

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.3.0:generate (default-cli) on project standalone-pom: 
  Execution default-cli of goal org.apache.maven.plugins:maven-archetype-plugin:3.3.0:generate failed: 
  An API incompatibility was encountered while executing org.apache.maven.plugins:maven-archetype-plugin:3.3.0:generate: 
  java.lang.NoSuchMethodError: 'org.apache.maven.project.ProjectBuildingRequest org.apache.maven.archetype.ArchetypeGenerationRequest.getProjectBuildingRequest()'
```

Why has this started failing now? Because on 22-Sep-2024 version  3.3.0 of [maven-archetype-plugin](https://maven.apache.org/archetype/maven-archetype-plugin/index.html) was released that introduced an incompatibility.  See [PR 215](https://github.com/apache/maven-archetype/pull/215)


## Workaround

The workaround is to use version 3.2.1 of `maven-archetype-plugin`:

```
mvn -U org.apache.maven.plugins:maven-archetype-plugin:3.2.1:generate  -DinteractiveMode=false \
    -DarchetypeGroupId=io.helidon.archetypes \
    -DarchetypeArtifactId=helidon-quickstart-mp \
    -DarchetypeVersion=4.1.1 \
    -DgroupId=io.helidon.examples \
    -DartifactId=helidon-quickstart-mp \
    -Dpackage=io.helidon.examples.quickstart.mp
```

Or even better us the [Helidon CLI](https://helidon.io/docs/v4/about/cli):

```
helidon init  --version 4.1.1 
```

--- a/.mvn/extensions.xml
+++ b/.mvn/extensions.xml
@@ -22,6 +22,6 @@
     <extension>
         <groupId>io.helidon.build-tools</groupId>
         <artifactId>helidon-build-cache-maven-extension</artifactId>
-        <version>4.0.11</version>
+        <version>4.0.12</version>
     </extension>
 </extensions>

--- a/applications/parent/pom.xml
+++ b/applications/parent/pom.xml
@@ -41,8 +41,8 @@
         <version.plugin.dependency>3.6.0</version.plugin.dependency>
         <version.plugin.exec>3.1.0</version.plugin.exec>
         <version.plugin.failsafe>3.1.2</version.plugin.failsafe>
-        <version.plugin.helidon>4.0.11</version.plugin.helidon>
-        <version.plugin.helidon-cli>4.0.11</version.plugin.helidon-cli>
+        <version.plugin.helidon>4.0.12</version.plugin.helidon>
+        <version.plugin.helidon-cli>4.0.12</version.plugin.helidon-cli>
         <version.plugin.jar>3.3.0</version.plugin.jar>
         <version.plugin.nativeimage>0.10.2</version.plugin.nativeimage>
         <version.plugin.os>1.5.0.Final</version.plugin.os>

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -168,7 +168,7 @@
                 <plugin>
                     <groupId>io.helidon.build-tools</groupId>
                     <artifactId>helidon-build-cache-maven-plugin</artifactId>
-                    <version>4.0.11</version>
+                    <version>4.0.12</version>
                 </plugin>
             </plugins>
         </pluginManagement>

--- a/pom.xml
+++ b/pom.xml
@@ -117,7 +117,7 @@
         <version.plugin.exec>3.1.0</version.plugin.exec>
         <version.plugin.failsafe>3.1.2</version.plugin.failsafe>
         <version.plugin.glassfish-copyright>2.3</version.plugin.glassfish-copyright>
-        <version.plugin.helidon-build-tools>4.0.11</version.plugin.helidon-build-tools>
+        <version.plugin.helidon-build-tools>4.0.12</version.plugin.helidon-build-tools>
         <version.plugin.hibernate-enhance>${version.lib.hibernate}</version.plugin.hibernate-enhance>
         <version.plugin.install>3.1.2</version.plugin.install>
         <version.plugin.jacoco>0.8.5</version.plugin.jacoco>

